From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ardun <ardunaline@gmail.com>
Date: Mon, 16 Jan 2023 22:26:18 +0300
Subject: [PATCH] Expose Item NBT


diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
index 90d7e8415f158c3176a705749f0aa36a729d3b38..deab77b95c089177aca9f8d2f4c861e9b19240e7 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
@@ -856,7 +856,16 @@ class CraftMetaItem implements ItemMeta, Damageable, Repairable, BlockDataMeta {
     boolean isEmpty() {
         return !(this.hasDisplayName() || this.hasLocalizedName() || this.hasEnchants() || (this.lore != null) || this.hasCustomModelData() || this.hasBlockData() || this.hasRepairCost() || !this.unhandledTags.isEmpty() || !this.persistentDataContainer.isEmpty() || this.hideFlag != 0 || this.isUnbreakable() || this.hasDamage() || this.hasAttributeModifiers() || this.hasPlaceableKeys() || this.hasDestroyableKeys()); // Paper - Implement an API for CanPlaceOn and CanDestroy NBT values
     }
-
+    //Legendary Start
+    @Override
+    public net.parkermc.legends.legendary.nbt.NBTTagCompound getTag() {
+        return this.internalTag;
+    }
+    @Override
+    public void setTag(net.parkermc.legends.legendary.nbt.NBTTagCompound tag) {
+        this.internalTag = (CompoundTag) tag;
+    }
+    //Legendary End
     // Paper start
     @Override
     public net.kyori.adventure.text.Component displayName() {
